pydantic/env_settings.py

```diff
@@ -38,6 +38,10 @@ def _build_environ(self) -> Dict[str, Optional[str]]:
         else:
             env_vars = {k.lower(): v for k, v in os.environ.items()}
 
+        env_file = self.__config__.env_file
```

Comment:
Please use a `Path` object here.

Eg.

```py
if self.__config__.env_file:
    env_path = Path(self.__config__.env_file)
    env_vars.update(self._read_from_env_file(env_path))
```

Reply:
I think it should also be possble to set `env_file` via a keyword argument to `__init__`, e.g. `_env_file`.

The keyword argument should take precedence over the config argument including if `_env_file=None`

Reply:
How does `_env_file=None` override the config argument without making it a required argument? i.e. How are we differentiating between not passing in `_env_file` and passing in `_env_file=None`? Potentially we need the disable functionality to be `_env_file=False`, no?

Reply:
Generally people do this using a "sentinel" value.

Something like

```py
env_file_sentinel = object()
...
def __init__(..., _env_file: Union[None, str, Path] = env_file_sentinel):
    ...
    if _env_file != env_file_sentinel:
        ...
```

Reply:
I think you could also just use `Ellipsis` (aka `...`) for this purpose here.

Reply:
ellipsis is fine with me.

Reply:
Maybe I'm doing something wrong, but `Ellipsis` doesn't seem to play well with `typing`.

Reply:
You'll need `#  type: ignore`, otherwise you could use the sentinel approach with either:
* `env_file_sentinel: str = object()  # type: ignore`
* or `env_file_sentinel = str(object())`

Reply:
or you could do

```py
from pathlib import Path
from typing import Union

EllipsisType = type(...)

def foobar(a: Union[None, str, Path, EllipsisType] = ...) -> None:
    return None
```